src/nacl/pwhash/__init__.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+from nacl.exceptions import InvalidkeyError
+
+from . import argon2, argon2i, argon2id, scrypt
+
+STRPREFIX = argon2id.STRPREFIX
+
+PWHASH_SIZE = argon2.PWHASH_SIZE
+SALTBYTES = argon2.SALTBYTES
+BYTES_MAX = argon2.BYTES_MAX
+BYTES_MIN = argon2.BYTES_MIN
+
+assert argon2.ALG_ARGON2_DEFAULT == argon2.ALG_ARGON2ID13
+# since version 1.0.15 of libsodium
+
+MEMLIMIT_MAX = argon2id.MEMLIMIT_MAX
```

Comment:
I don't think we should expose these as `nacl.pw_hash.MEMLIMIT_MAX`. libsodium has made argon2id the default, but it still uses namespaced functions like `crypto_pwhash_argon2id_memlimit_max` right?

Reply:
These shims follow libsodium's logic, which defines shims in the upper   crypto_pwhash_ namespace to "lift" functions and constants  from the namespace of the crypto_pwhash_${mech}_ choosen as the default one.

An example: ``include/sodium/crypto_pwhash.h`` 
```
#define crypto_pwhash_MEMLIMIT_MIN crypto_pwhash_argon2id_MEMLIMIT_MIN
SODIUM_EXPORT
size_t crypto_pwhash_memlimit_min(void);
```
If you feel strong against this, we could avoid defining the top level shims, and the verify() dispatcher, but they seem useful.

Reply:
An example of using the pwhash module level constants would be:
```
>>> from nacl import pwhash
>>> from nacl.pwhash import scrypt
>>> pwhash.str(b'mypass', opslimit=pwhash.OPSLIMIT_INTERACTIVE, memlimit=pwhash.MEMLIMIT_INTERACTIVE)
b'$argon2id$v=19$m=65536,t=2,p=1$o8i1RIGIEPdEbskygmZlng$8JaX4WcDZcgzv1IEV/G+NJZafRQS213YiKLk9zCEI5g'
>>> scrypt.str(b'mypass', opslimit=scrypt.OPSLIMIT_INTERACTIVE, memlimit=scrypt.MEMLIMIT_INTERACTIVE)
b'$7$C6..../....qdeSPD.9L.6lTQD5xLbQAzq4AWkVxsVWJUgxju.hs6B$Zh3by6ga7OZiLlVVh0MeKTRCzDV60XWg2sk9NFd.f1/'
>>>
```

In which both the pwhash and the pwhash.scrypy expose the same API.